pkg/kubelet/status/manager.go

```diff
@@ -338,6 +340,20 @@ func (m *manager) syncBatch() {
 	}
 }
 
+func (m *manager) podExists(uid types.UID) bool {
+	m.podStatusesLock.RLock()
+	defer m.podStatusesLock.RUnlock()
+	_, hasPod := m.podStatuses[uid]
+	return hasPod
+}
+
+func (m *manager) syncPodCheckExistence(uid types.UID, status versionedPodStatus) {
```

Comment:
nit: This function checks existence and then runs a syncPod. Shall we rename this to `checkExistenceSyncPodIfNeeded()`?